### PR TITLE
Completed D400 for multiple folders

### PR DESCRIPTION
**Pull request by bdsoha**
Completed D400 for:
- `airflow/__init__.py`
- `airflow/__main__.py`
- `airflow/dag_processing/*`
- `airflow/logging_config.py`
- `airflow/plugins_manager.py`
- `airflow/providers_manager.py`
- `airflow/sentry.py`
- `airflow/settings.py`
- `airflow/stats.py`
- `airflow/templates.py`
- `airflow/typing_compat.py`

Related to #10742 .

### CODE REVIEW

File: airflow/providers_manager.py [hunk 1]
```diff
@@ -542,9 +555,11 @@ def _discover_hooks_from_connection_types(
         provider: ProviderInfo,
     ):
         """
-        Discover  hooks from the "connection-types" property. This is new, better method that replaces
-        discovery from hook-class-names as it allows to lazy import individual Hook classes when they
-        are accessed. The "connection-types" keeps information about both - connection type and class
+        Discover  hooks from the "connection-types" property.
```
Comment (uranusjr) [inline airflow/providers_manager.py]:
```suggestion
        Discover hooks from the "connection-types" property.
```

File: airflow/providers_manager.py [hunk 2]
```diff
@@ -446,11 +457,13 @@ def initialize_providers_auth_backends(self):
 
     def _discover_all_providers_from_packages(self) -> None:
         """
-        Discovers all providers by scanning packages installed. The list of providers should be returned
-        via the 'apache_airflow_provider' entrypoint as a dictionary conforming to the
-        'airflow/provider_info.schema.json' schema. Note that the schema is different at runtime
-        than provider.yaml.schema.json. The development version of provider schema is more strict and changes
-        together with the code. The runtime version is more relaxed (allows for additional properties)
+        Discovers all providers by scanning packages installed.
```
Comment (uranusjr) [inline airflow/providers_manager.py]:
```suggestion
        Discover all providers by scanning packages installed.
```

File: airflow/providers_manager.py [hunk 3]
```diff
@@ -595,9 +610,10 @@ def _discover_hooks_from_hook_class_names(
         provider_uses_connection_types: bool,
     ):
         """
-        Discovers hooks from "hook-class-names' property. This property is deprecated but we should
-        support it in Airflow 2. The hook-class-names array contained just Hook names without connection
-        type, therefore we need to import all those classes immediately to know which connection types
+        Discovers hooks from "hook-class-names' property.
```
Comment (uranusjr) [inline airflow/providers_manager.py]:
```suggestion
        Discover hooks from "hook-class-names' property.

```

File: airflow/providers_manager.py [hunk 4]
```diff
@@ -661,7 +677,7 @@ def _discover_hooks_from_hook_class_names(
             )
 
     def _discover_hooks(self) -> None:
-        """Retrieves all connections defined in the providers via Hooks"""
+        """Retrieves all connections defined in the providers via Hooks."""
```
Comment (uranusjr) [inline airflow/providers_manager.py]:
```suggestion
        """Retrieve all connections defined in the providers via Hooks."""
```

File: airflow/providers_manager.py [hunk 5]
```diff
@@ -721,7 +737,7 @@ def _add_taskflow_decorator(self, name, decorator_class_name: str, provider_pack
 
     @staticmethod
     def _get_attr(obj: Any, attr_name: str):
-        """Retrieves attributes of an object, or warns if not found"""
+        """Retrieves attributes of an object, or warns if not found."""
```
Comment (uranusjr) [inline airflow/providers_manager.py]:
```suggestion
        """Retrieve attributes of an object, or warn if not found."""
```

File: airflow/providers_manager.py [hunk 6]
```diff
@@ -735,10 +751,11 @@ def _import_hook(
         package_name: str | None = None,
     ) -> HookInfo | None:
         """
-        Imports hook and retrieves hook information. Either connection_type (for lazy loading)
-        or hook_class_name must be set - but not both). Only needs package_name if hook_class_name is
-        passed (for lazy loading, package_name is retrieved from _connection_type_class_provider_dict
-        together with hook_class_name).
+        Imports hook and retrieves hook information.
```
Comment (uranusjr) [inline airflow/providers_manager.py]:
```suggestion
        Import hook and retrieve hook information.
```

File: airflow/providers_manager.py [hunk 7]
```diff
@@ -883,31 +900,31 @@ def _add_customized_fields(self, package_name: str, hook_class: type, customized
             )
 
     def _discover_extra_links(self) -> None:
-        """Retrieves all extra links defined in the providers"""
+        """Retrieves all extra links defined in the providers."""
```
Comment (uranusjr) [inline airflow/providers_manager.py:904]:
```suggestion
        """Retrieve all extra links defined in the providers."""
```

File: airflow/providers_manager.py [hunk 8]
```diff
@@ -883,31 +900,31 @@ def _add_customized_fields(self, package_name: str, hook_class: type, customized
...
         for provider_package, provider in self._provider_dict.items():
             if provider.data.get("extra-links"):
                 for extra_link_class_name in provider.data["extra-links"]:
                     if _sanity_check(provider_package, extra_link_class_name, provider):
                         self._extra_link_class_name_set.add(extra_link_class_name)
 
     def _discover_logging(self) -> None:
-        """Retrieves all logging defined in the providers"""
+        """Retrieves all logging defined in the providers."""
```
Comment (uranusjr) [inline airflow/providers_manager.py]:
```suggestion
        """Retrieve all logging defined in the providers."""
```

File: airflow/providers_manager.py [hunk 9]
```diff
@@ -883,31 +900,31 @@ def _add_customized_fields(self, package_name: str, hook_class: type, customized
...
         for provider_package, provider in self._provider_dict.items():
             if provider.data.get("logging"):
                 for logging_class_name in provider.data["logging"]:
                     if _sanity_check(provider_package, logging_class_name, provider):
                         self._logging_class_name_set.add(logging_class_name)
 
     def _discover_secrets_backends(self) -> None:
-        """Retrieves all secrets backends defined in the providers"""
+        """Retrieves all secrets backends defined in the providers."""
```
Comment (uranusjr) [inline airflow/providers_manager.py]:
```suggestion
        """Retrieve all secrets backends defined in the providers."""
```

File: airflow/providers_manager.py [hunk 10]
```diff
@@ -883,31 +900,31 @@ def _add_customized_fields(self, package_name: str, hook_class: type, customized
...
         for provider_package, provider in self._provider_dict.items():
             if provider.data.get("secrets-backends"):
                 for secrets_backends_class_name in provider.data["secrets-backends"]:
                     if _sanity_check(provider_package, secrets_backends_class_name, provider):
                         self._secrets_backend_class_name_set.add(secrets_backends_class_name)
 
     def _discover_auth_backends(self) -> None:
-        """Retrieves all API auth backends defined in the providers"""
+        """Retrieves all API auth backends defined in the providers."""
```
Comment (uranusjr) [inline airflow/providers_manager.py]:
```suggestion
        """Retrieve all API auth backends defined in the providers."""
```

File: airflow/providers_manager.py [hunk 11]
```diff
@@ -923,8 +940,8 @@ def providers(self) -> dict[str, ProviderInfo]:
     @property
     def hooks(self) -> MutableMapping[str, HookInfo | None]:
         """
-        Returns dictionary of connection_type-to-hook mapping. Note that the dict can contain
-        None values if a hook discovered cannot be imported!
+        Returns dictionary of connection_type-to-hook mapping.
```
Comment (uranusjr) [inline airflow/providers_manager.py]:
```suggestion
        Return dictionary of connection_type-to-hook mapping.

```

File: airflow/settings.py [hunk 12]
```diff
@@ -467,14 +470,11 @@ def validate_session():
...
-    """
-    Any additional configuration (register callback) for airflow.utils.action_loggers
-    module
-    """
+    """Any additional configuration (register callback) for airflow.utils.action_loggers module."""
 
 
 def prepare_syspath():
-    """Ensures that certain subfolders of AIRFLOW_HOME are on the classpath"""
+    """Ensures that certain subfolders of AIRFLOW_HOME are on the classpath."""
```
Comment (uranusjr) [inline airflow/settings.py]:
```suggestion
    """Ensure certain subfolders of AIRFLOW_HOME are on the classpath."""
```

File: airflow/plugins_manager.py [hunk 13]
```diff
@@ -378,7 +378,7 @@ def initialize_ti_deps_plugins():
 
 
 def initialize_extra_operators_links_plugins():
-    """Creates modules for loaded extension from extra operators links plugins"""
+    """Creates modules for loaded extension from extra operators links plugins."""
```
Comment (uranusjr) [inline airflow/plugins_manager.py]:
```suggestion
    """Create modules for loaded extension from extra operators links plugins."""
```

File: airflow/plugins_manager.py [hunk 14]
```diff
@@ -357,7 +357,7 @@ def initialize_web_ui_plugins():
 
 
 def initialize_ti_deps_plugins():
-    """Creates modules for loaded extension from custom task instance dependency rule plugins"""
+    """Creates modules for loaded extension from custom task instance dependency rule plugins."""
```
Comment (uranusjr) [inline airflow/plugins_manager.py]:
```suggestion
    """Create modules for loaded extension from custom task instance dependency rule plugins."""
```

**Comment by uranusjr** [comment]
@bdsoha You can bulk-apply all suggestions from the “files” tab instead of adding individual commits.

**Comment by bdsoha** [comment]
> @bdsoha You can bulk-apply all suggestions from the “files” tab instead of adding individual commits.

Thanks for the tip, will use it next time.

**Comment by potiuk** [comment]
sounds like new warning generated when importing providers. ?

**Comment by bdsoha** [comment]
@potiuk Any suggestions? Rebase?

**Comment by potiuk** [comment]
Or rather fix it. Not sure where it came from though - it should not happen normally but might be triggered by some of your changes - if it does not happen in main, just follow the instructions printed with the error.

**Comment by potiuk** [comment]
Needs rebase I am afraid (conflicts after the merge of previous D400 change) @bdsoha

**Comment by bdsoha** [comment]
@potiuk All green...

**Review by potiuk — APPROVED** [review]

**Pull request merged**